### PR TITLE
Run tests and linting with GitHub Actions

### PR DESCRIPTION
**Pull request by alexgleith**
This PR does a few things:

- Removes requirements-test.txt
- Removes DEV Dockerfile
- Removes Travis builds
- Adds GitHub Actions linting/testing (tested and works, bar one test which is now skipped)
- Adds GitHub Actions Docker builds (untested, only on the develop branch currently)
- Same Dockerfile for DEV and PROD (minor difference in setup.py usage for installation)
- Same Docker Compose file for GitHub Actions and local test/linting.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+env:
+  IMAGE_NAME: opendatacube/dashboard
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
```
Comment (omad) [inline .github/workflows/main.yml]: They recommend [actions/checkout@v2](https://github.com/actions/checkout) now

**Review by omad — COMMENTED** [review]
Looks good to me.

Merge first, ask questions later.

**Review by jeremyh — COMMENTED** [review]
Looks good to me too

File: .github/workflows/main.yml (hunk 1 above)
Comment (alexgleith) [reply .github/workflows/main.yml]: I'll update this.

**Review by omad — DISMISSED** [review]

**Comment by whatnick** [comment]
Could we have _version.py populated ? See -> https://github.com/opendatacube/datacube-explorer/pull/98

**Comment by alexgleith** [comment]
Sorry, had to force push so that rebasing works...

**Review by jeremyh — APPROVED** [review]

**Review by whatnick — APPROVED** [review]
👍

**Pull request merged**